File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let max = 0;
        let buy = Infinity;

        for (let i = 0; i < prices.length; i++) {
            if (prices[i] - buy < 0) {
                buy = prices[i];
            }
            else {
                max = Math.max(max, prices[i]- buy);
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        curr_profit = 0
        max_profit = 0
        buy_price = prices[0]

        for price in prices:
            if price - buy_price > curr_profit:
                curr_profit = price - buy_price
            else:
                max_profit += curr_profit
                curr_profit = 0
                buy_price = price
        max_profit += curr_profit

        return max_profit